src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs

```diff
@@ -132,7 +156,18 @@ public static class TracerProviderBuilderExtensions
             {
                 var instrumentation = sp.GetRequiredService<StackExchangeRedisInstrumentation>();
 
-                connection ??= sp.GetService<IConnectionMultiplexer>();
+                if (connection is not null)
+                {
+                    if (serviceKey is null)
+                    {
+                        connection = sp.GetService<IConnectionMultiplexer>();
+                    }
+                    else
+                    {
+                        connection = sp.GetKeyedService<IConnectionMultiplexer>(serviceKey);
+                    }
+                }
+
 
                 if (connection != null)
```

Comment:
```suggestion
                if (connection is not null)
```

Might as well do the null checks in the same method in a consistent way.